src/routes/webhooks.ts

```diff
@@ -0,0 +1,304 @@
+import express from 'express'
+import { Constants } from '../shared/constants';
+import { authCommons } from '../shared/auth_commons';
+import { User } from '../models/user/user';
+import { WebhookEvent } from '../enums/webhookEvents';
+import { WebhookEventObjectType } from '../enums/webhookServiceObjectType';
+import { DatabaseService, databaseService } from '../shared/database_service';
+import { coreService } from '../shared/core_service';
+import { ObjectId } from 'mongodb';
+import { Connection } from '../models/connection/connection';
+import { getJiraIssueById } from '../shared/services_config_functions';
+import { Service } from 'ts-node';
+import { SyncedService } from '../models/connection/config/synced_service';
+import { timeStamp } from 'console';
+import { WebhookEventData } from '../models/connection/config/webhook_event_data';
+
+
+const router = express.Router({ mergeParams: true });
+router.use(express.urlencoded({ extended: false }));
+router.use(express.json());
+
+router.use((req, res, next) => {
+    console.log(`webhook router calls at time: ${Date().toString()}`);
+
+    // // For CORS policy
+    // res.append('Access-Control-Allow-Origin', ['*']);
+    // res.append('Access-Control-Allow-Methods', 'GET,PUT,POST,DELETE');
+    // res.append('Access-Control-Allow-Headers', 'Origin,X-Requested-With,Content-Type,Accept,Authorization,sentry-trace');
+    next();
+});
+
+
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+let t2tLib: any;
+if (Constants.isCommercialVersion) {
+    t2tLib = require('@timer2ticket/timer2ticket-backend-library');
+}
+
+
+router.post('/jira/:connectionId', async (req, res) => {
+    res.sendStatus(200)
+    if (!req.body)
+        return
+    let connectionId
+    try {
+        connectionId = new ObjectId(req.params.connectionId);
+    } catch (err: any) {
+        // console.log('unable to get connection Id')
+        return
+    }
+    //get webhook type
+    const body = req.body
+    const event = _getJiraWebhookEvent(body)
+    if (event === WebhookEvent.Error) {
+        // console.log('unable to read event')
+        return
+    }
+    const eventObject = _getJiraEventObjectType(body)
+    if (eventObject === WebhookEventObjectType.Error) {
+        // console.log('unable to read object from webhook')
+        return
+    }
+    //get user
+    const accountId = _getJiraUserId(body, eventObject)
+    if (!accountId)
+        return
+    //get connection
+    const connection = await databaseService.getConnectionById(connectionId)
+    if (!connection) {
+        // console.log('unable to get COnnection')
+        return
+    }
+    //check subscription
+    const subscriptionAllowsWebhooks = await _checkSubscriptionForWebhook(connection)
+    if (!subscriptionAllowsWebhooks) {
```

Comment:
K zamyšlení: nebylo by dobré nějak webhooky v tomto případě stopnout...

Reply:
Vždyť ten webhook stopnu. Asi nerozumím

Reply:
Myslel jsem - jestli se nepletu tak ty v  Toggl (a v dalších službách) registruješ webhooky. Tak jestli bychom je neměli smazat, když už je uživatel nemůže použít. Může to být docela zátěž na systém, když přijde webhook, který jen ignorujeme...

Reply:
Dobrý point, řešení je daleko od triviálního